rflx/model/message.py

```diff
@@ -1569,6 +1569,52 @@ def proven(self, skip_proof: bool = False) -> "DerivedMessage":
         return copy(self)
 
 
+def replace_message_attributes(
+    message: AbstractMessage,
+) -> Tuple[AbstractMessage, List[Optional[Location]]]:
+    """
+    Replace all occurences of the message attribute Message'Size by the equivalent term
+    Message'Last - Message'First + 1. Then replace all occurences of the message attribute
+    Message'First by a reference to the first field of the message. Return the new message and
+    a list of locations where Message'Last or Message'Size have been used.
+    """
+    first_field = message.outgoing(INITIAL)[0].target
+    locations = []
+
+    def replace(expression: expr.Expr) -> expr.Expr:
+        if (
+            not isinstance(expression, expr.Attribute)
+            or not isinstance(expression.prefix, expr.Variable)
+            or expression.prefix.name != "Message"
+        ):
+            return expression
+        if isinstance(expression, expr.First):
+            return expr.First(expr.ID(first_field.identifier, location=expression.location))
```

Comment:
`ID` should be used directly, not via the `expr` module.

Reply:
Changed in 877fd78d.